### ISSUE TITLE
SyncTex broken on 4.1.1

### ISSUE THREAD

**Issue by wixaw** [overleaf/overleaf#1150]
Hello, 
I just upgraded to 4.1.1, and my users are complaining that SyncTex feature is no longer working. The arrows are spinning in circles.
Thank you.

![image](https://github.com/overleaf/overleaf/assets/7153125/3bec6dc2-7de5-4757-b284-187cc75f0be5)

![image](https://github.com/overleaf/overleaf/assets/7153125/d75915c6-5e1c-4161-bfbb-0de4945dbfd0)

**Comment by wixaw** [comment]
Hello,
I've discovered some new information. When I disable HTTP2 on my front-end proxy, it works.

What's surprising is that it's only the XHR request for synctex that has this issue; everything else is functioning correctly. This particular request remains in a pending state until it times out. If we directly access https://url/project/xxxx/sync/code?file=Rapport%2F.%2FRapport.tex&line=752&column=826, we can see that the request takes 20/30 seconds to respond in HTTP2, while it's instant in HTTP1.

Perhaps there's an issue with the Node.js update? In any case, it would be great to find the reason for this, as it's slowing down page loading.

Please let me know if you need more information or if there's anything specific you'd like me to check.

**Comment by wixaw** [comment]
Hello @mserranom @aeaton-overleaf 
We just realized that without HTTP2, our SSL stream, decrypted by the firewall for the WAF, is causing issues with the pdf.js output. When we reactivate it, it works, but we lose the synctex functionality.
Do you have any suggestions? Would you like additional logs?
Thanks in advance.
( Sorry for the mention, but I haven't heard anything about this problem since September )

**Comment by mserranom** [comment]
Hi @wixaw. We haven't received any report describing problems similar to this, seems to be very specific of your networking configuration. 

Do you know if the problem started only after upgrading to `4.1.x`, and wether it was working as expected with previous versions?

**Comment by wixaw** [comment]
Yes, it was working fine before in 4.0.x. Our users complained as soon as I upgraded to 4.1.1. My configuration is simple: a front-end Apache proxypass on port 443 that redirects to our Docker application server. For your information, if I don't go through SSL decryption, it's the same, SyncTeX doesn't work either with http2.

**Comment by das7pad** [comment]
Hello,

Thanks for writing this up. It looks like a setup specific issue to me. The sync to code and sync to PDF features are working in the latest release.
You might also be interested in the latest addition to our release process: We've added automated e2e tests that also [cover the SyncTeX feature](https://github.com/overleaf/overleaf/blob/169723f492eb083ae3ebd5c6038b04a7a0bdf4a7/server-ce/test/sandboxed-compiles.spec.ts#L63). :)

Greetings,
Jakob

**Comment by das7pad** [comment]
I'm afraid #952 is exceeding the scope of support that we can give for Overleaf Community Edition.

**Comment by wixaw** [comment]
Hello @das7pad  @mserranom 

I have some new tests to report to you!

On the same machine, I have an Apache server and the Overleaf Docker containers (this reproduces my production setup where I have a frontal proxy on one server and the application on another).

Apache configuration:
```
<VirtualHost _default_:443>
    ServerName wwwap1r8test.irit.fr
    LogLevel warn

    SSLEngine on
    SSLProxyEngine On
    Protocols h2 http/1.1

    RemoteIPHeader X-Forwarded-For

    RewriteEngine On
    RewriteCond %{HTTP:Upgrade} =websocket
    RewriteRule /(.*)           ws://127.0.0.1:5000/$1 [P,L]
    RewriteCond %{HTTP:Upgrade} !=websocket
    RewriteRule /(.*)           http://127.0.0.1:5000/$1 [P,L]
</VirtualHost>
```
And in the Docker container, here is the Nginx log when I try to use synctex:


```
==> /var/log/nginx/error.log <==
2024/09/13 16:27:58 [error] 420#420: *438 upstream timed out (110: Unknown error) while reading response header from upstream, client: 10.10.51.1, server: _, request: "GET /project/65f81a493d18563068f10140/sync/pdf?page=1&h=281.69&v=137.57 HTTP/1.1", upstream: "http://127.0.0.1:4000/project/65f81a493d18563068f10140/sync/pdf?page=1&h=281.69&v=137.57", host: "127.0.0.1:5000", referrer: "https://wwwap1r8test.irit.fr/project/65f81a493d18563068f10140"
```

If i disable h2, that work!
So there is no firewall blocking in this case.
I don't know what changed on the Nginx side, except that it is no longer port 3000 but 4000.

Thanks

**Comment by das7pad** [comment]
Thanks for the update, I'm glad to hear that you managed to get 5.x running.

I'm afraid we do officially support running Apache as reverse proxy in front of Server Pro. Thanks for sharing the snippet, it might be useful for other members of the community still.

Regarding the ports, the web-api service is now running on port 3000 and the user facing web service runs on port 4000.
This was shipped with Server Pro 4.2.0 in November 2023: https://github.com/overleaf/overleaf/wiki/Release-Notes--4.x.x#server-pro-420
